### ISSUE TITLE
S2365: Fix VB.Net implementation

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#7360]
The description of the visual basic variant for S2365 is correct on RSPEC, but the implementation is not.

The [implementation](https://github.com/SonarSource/sonar-dotnet/blob/16fcbbce57c68c707a2f33a0658eb139cc914991/analyzers/src/SonarAnalyzer.VisualBasic/Rules/PropertyWithArrayType.cs#L24) checks almost only that the property's type is `IArraySymbol`, and not what happens inside the body of the getter/setter:

```csharp
// Current implementation
if (property.ImplementsClause is null
    && !property.Modifiers.Any(x => x.IsKind(SyntaxKind.OverridesKeyword))
    && c.SemanticModel.GetDeclaredSymbol(property) is IPropertySymbol symbol
    && !symbol.IsAutoProperty()
    && symbol.Type is IArrayTypeSymbol)
``` 

There is no reference to `Clone` or the other methods that should be avoided, which even simple cases raise. Example:

```vb
Public Interface IBase
    Property Evil As Byte()    ' Noncompliant: passes all the checks above
End Interface
```

## Expected behavior

Refactor the visual basic implementation to be semantically the same as the C# implementation.

**Comment by gregory-paidis-sonarsource** [comment]
setting priority to high, because it is SonarWay

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [NET-1771](https://sonarsource.atlassian.net/browse/NET-1771)

[NET-1771]: https://sonarsource.atlassian.net/browse/NET-1771?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
